src/libs/elektra/keyvalue.c

```diff
@@ -526,6 +528,18 @@ ssize_t keySetRaw (Key * key, const void * newBinary, size_t dataSize)
 	if (key->data.v)
 	{
 		char * previous = key->data.v;
+
+		if (dataInMmap)
+		{
+			clear_bit (key->flags, KEY_FLAG_MMAP_DATA);
+			key->data.v = elektraMalloc (key->dataSize);
+			if (key->data.v == NULL) return -1;
+		}
+		else
+		{
+			if (-1 == elektraRealloc ((void **) &key->data.v, key->dataSize)) return -1;
```

Comment:
good to have the check here, but it is now inconsistent with the other places where the return value of elektraRealloc is not checked? (Can you add the check everywhere?)

Reply:
I added the check on every elektraRealloc where I already touched the source. I think adding the check on **every** realloc or malloc would be better in a separate issue/PR?

Reply:
No, if it is consistent for every realloc and malloc you touched it is perfect for this PR.